drivers/hmis/app/graphql/mutations/create_duplicate_form_definition.rb

```diff
@@ -0,0 +1,67 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class CreateDuplicateFormDefinition < CleanBaseMutation
+    argument :identifier, String, required: true
+
+    field :form_identifier, Types::Forms::FormIdentifier, null: true
+
+    def resolve(identifier:)
+      # Choose the most recent Published or Retired definition to duplicate
+      definition_to_duplicate = Hmis::Form::Definition.
+        where(identifier: identifier).
+        published_or_retired.
+        order(version: :desc).first
+
+      raise 'not found' unless definition_to_duplicate
+
+      access_denied! unless current_user.can_manage_forms_for_role?(definition_to_duplicate.role)
+
+      # Drop all custom_field_key mappings on definition structure
+      cleaned_definition_json = remove_custom_field_mappings(definition_to_duplicate)
```

Comment:
This is actually in-place-modifying the `definition_to_duplicate.definition` attribute, and then it just doesn't save changes on `definition_to_duplicate` to the db, right?

Maybe code-wise it would be clearer/safer to clone this definition first before `walk_definition_nodes`? Maybe this takes up more space in memory though?

Reply:
Yeah fair point, I'll throw in a `dup`!